hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java

```diff
@@ -138,14 +139,36 @@ public Collection<ConsensusNode> getNodes() {
         var totalStake = new AtomicLong(0L);
         var nodes = new TreeSet<ConsensusNode>();
         var nodeStakes = new HashMap<Long, NodeStake>();
+        ConsensusMode consensusMode = mirrorProperties.getConsensusMode();
 
         nodeStakeRepository.findLatest().forEach(nodeStake -> {
+            if (consensusMode == ConsensusMode.EQUAL) {
+              nodeStake.setStake(1L);
+            }
             totalStake.addAndGet(nodeStake.getStake());
             nodeStakes.put(nodeStake.getNodeId(), nodeStake);
         });
 
         // For partial mirror nodes, node stake will provide a more accurate count
-        long nodeCount = !nodeStakes.isEmpty() ? nodeStakes.size() : addressBook.getNodeCount();
+        long nodeCount = (consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK || nodeStakes.isEmpty()) ?
+            addressBook.getNodeCount() : nodeStakes.size();
+
+        // if only including address book nodes in stake count, reset and readd only address book nodes' stakes.
+        if (consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK) {
+          long oldTotalStake = totalStake.get();
+          totalStake.set(0L);
+          addressBook.getEntries().forEach(e -> {
+              var nodeStake = nodeStakes.get(e.getNodeId());
+              if (nodeStake != null) {
+                totalStake.addAndGet(nodeStake.getStake());
+              }
+          });
+          long newTotalStake = totalStake.get();
+          if (oldTotalStake != newTotalStake) {
+            log.warn("Using address book with {}/{} nodes for node staking, total stake of {}/{}",
+                addressBook.getNodeCount(), nodeStakes.size(), newTotalStake, oldTotalStake);
```

Comment:
I think it would be better and simpler to just warn if the size differs. Also, would be better for researching to print the timestamps of each. Just pick one timestamp from nodestakes.

```suggestion
          if (addressBook.getNodeCount() != nodeStakes.size()) {
            log.warn("Using address book {} with {} nodes and node stake {} with {} nodes",
                addressBook.getConsensusTimestamp(), addressBook.getNodeCount(), nodeStakeTimestamp, nodeStakes.size());
```

Reply:
Done.